src/detect.ts

```diff
@@ -10,23 +10,31 @@ export function detectImageSupport(url: string) {
   return detectMediaSupport('img', url);
 }
 
-export async function findAvailableImageURL(srcset: ArrayLike<string>) {
-  for (const url of Array.from(srcset)) {
-    if (await detectImageSupport(url)) {
-      return url;
-    }
-  }
-  throw new Error('unavailable');
+export function findAvailableImageURL(srcset: ArrayLike<string>, timeout?: number) {
+  const promises = Array.from(srcset).map((url) => detectImageSupport(url));
+  const id =
+    timeout &&
+    setTimeout(() => {
```

Comment:
the error won't be catched from this block since settimeout() is a macro task which runs in main loop.